tests/ui_tools/test_boxes.py

```diff
@@ -99,8 +99,10 @@ def test_autocomplete_mentions(self, write_box, users_fixture,
         assert typeahead_string == required_typeahead
 
     @pytest.mark.parametrize('text, state, required_typeahead', [
-        ('#Stream', 0, '#**Stream 1**'),
-        ('#Stream', 1, '#**Stream 2**'),
+        ('#Stream', 0, '#**Some general stream**'),
+        ('#Stream', 1, '#**Secret stream**'),
+        ('#Stream', 2, '#**Stream 1**'),
+        ('#Stream', 3, '#**Stream 2**'),
         ('#S', 0, '#**Some general stream**'),
         ('#S', 1, '#**Secret stream**'),
```

Comment:
Wouldn't it be more desirable if it matches a stream starting with some text, and only then do a full text search?

Reply:
I'd like to know what @neiljp thinks about this.

Reply:
Intuitively, I would expect to get search results which match with the starting text first and later on those which match with intermediate text. Maybe a full text search can be implemented as a secondary sort.